src/Microsoft.DotNet.CodeAnalysis/Analyzers/ResourceUsageAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+    public class ResourceUsageAnalyzer : DiagnosticAnalyzer
+    {
+        private const string Title = "Invalid SR.Format call";
+        private const string Description = "The SR.Format call should be removed.";
+        private const string AnalyzerName = "ResourceUsageAnalyzer";
+
+        private static DiagnosticDescriptor InvalidSRFormatCall = new DiagnosticDescriptor(DiagnosticIds.BCL1000.ToString(), Title, "", AnalyzerName, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(InvalidSRFormatCall); } }
+
+        INamedTypeSymbol SRSymbol { get; set; }
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(StartCompile);
+        }
+
+
+        private void StartCompile(CompilationStartAnalysisContext obj)
+        {
+            SRSymbol = obj.Compilation.GetTypeByMetadataName("System.SR");
+            if (SRSymbol != null)
+            {
+                obj.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.InvocationExpression);
+            }
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext obj)
+        {
+            InvocationExpressionSyntax invokeExpr = obj.Node as InvocationExpressionSyntax;
+
+            if (invokeExpr == null) return;
+
+            MemberAccessExpressionSyntax memberAccessExpr = invokeExpr.Expression as MemberAccessExpressionSyntax;
+            if (memberAccessExpr == null) return;
+
+            IMethodSymbol memberSymbol = obj.SemanticModel.GetSymbolInfo(memberAccessExpr).Symbol as IMethodSymbol;
+            if (memberSymbol == null) return;
+
+            if (memberSymbol.Name.Equals("Format") && memberSymbol.ContainingType.Equals(SRSymbol))
+            {
+                bool isValidCall = false;
+                if (memberSymbol.Parameters.Length > 1
+                    && memberSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                {
+                    // The method could be called directly by passing in an object array.
+                    if (memberSymbol.Parameters[1].IsParams
+                    && memberSymbol.Parameters[1].Type is IArrayTypeSymbol
+                    && ((IArrayTypeSymbol)memberSymbol.Parameters[1].Type).ElementType.SpecialType == SpecialType.System_Object)
+                    {
+                        isValidCall = true;
+                    }
+
+                    // 2. Method could be called with variable arguments of type object.
+                    else
+                    {
+                        isValidCall = true;
+                        for (int i = 1; i < memberSymbol.Parameters.Length; i++)
+                        {
+                            if (memberSymbol.Parameters[i].Type.SpecialType != SpecialType.System_Object)
+                                isValidCall = false;
```

Comment:
As of now we could have made the condition simply if Parameters.Length >=2 but I just wanted to make it tighter and use this instead. Do you see any issue with it?